contrib/babelfishpg_tsql/src/linked_servers.c

```diff
@@ -94,6 +94,49 @@ linked_server_msg_handler(LinkedServerProcess lsproc, int error_code, int state,
 	return 0;
 }
 
+/*
+ * Helper function to remove all occurrences
+ * of a substring from a source string
+ */
+static char *
+remove_substr(char *src, const char *substr)
+{
+	char *str;
+	char *result = pstrdup("");
+	int substr_len = strlen(substr);
+
+	if (!strstr(src, substr))
+		return src;
+
+	while ((str = strstr(src, substr)) != NULL)
+	{
+		char *tmp = psprintf("%s%s", result, pnstrdup(src, str - src));
+		if (result)
+		{
+			pfree(result);
+			result = tmp;
+		}
+
+		/* skip substr part */
+		src = str + substr_len;
```

Comment:
we need to be careful that str is NULL terminated

Reply:
Yes using pnstrdup now, so string will always be NULL terminated.